File: jadx-reference/rock/receiver/hhymfsyujsj.java
package com.storm.safe.rock.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import p000.AbstractC1120qr;
import p000.al1;
import p000.t60;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class hhymfsyujsj extends BroadcastReceiver {

    /* renamed from: a0 */
    public static final C0271a0 f52289a0 = new C0271a0(null);

    /* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
    /* renamed from: com.storm.safe.rock.receiver.hhymfsyujsj$a0 */
    public static final class C0271a0 {
        public /* synthetic */ C0271a0(AbstractC1120qr abstractC1120qr) {
            this();
        }

        public static /* synthetic */ void scheduleHeartbeat$default(C0271a0 c0271a0, Context context, long j, int i, Object obj) {
            if ((i & 2) != 0) {
                j = 0;
            }
            c0271a0.scheduleHeartbeat(context, j);
        }

        public static /* synthetic */ void scheduleNextPing$default(C0271a0 c0271a0, Context context, long j, int i, Object obj) {
            if ((i & 2) != 0) {
                j = 0;
            }
            c0271a0.scheduleNextPing(context, j);
        }

        public static /* synthetic */ void scheduleRecovery$default(C0271a0 c0271a0, Context context, long j, int i, int i2, Object obj) {
            if ((i2 & 4) != 0) {
                i = 0;
            }
            c0271a0.scheduleRecovery(context, j, i);
        }

        public final void cancelPing(Context context) {
            t60.m214695b6(context, "context");
        }

        public final void cancelRecovery(Context context) {
            t60.m214695b6(context, "context");
        }

        public final void scheduleDualRecovery(Context context) {
            t60.m214695b6(context, "context");
            scheduleHeartbeat$default(this, context, 0L, 2, null);
        }

        public final void scheduleHeartbeat(Context context, long j) {
            t60.m214695b6(context, "context");
        }

        public final void scheduleNextPing(Context context, long j) {
            t60.m214695b6(context, "context");
            scheduleHeartbeat(context, j);
        }

        public final void scheduleRecovery(Context context, long j, int i) {
            t60.m214695b6(context, "context");
            scheduleHeartbeat(context, j);
        }

        public final void startServices(Context context) {
            t60.m214695b6(context, "context");
            try {
                al1.f43714a5.getInstance(context).m209821a1();
            } catch (Exception e) {
                t60.m214705c6("hhymfsyujsj", "启动服务失败", e);
            }
            System.currentTimeMillis();
            C0271a0 c0271a0 = hhymfsyujsj.f52289a0;
        }

        private C0271a0() {
        }
    }

    @Override // android.content.BroadcastReceiver
    public void onReceive(Context context, Intent intent) {
        t60.m214695b6(context, "context");
        t60.m214702c3("hhymfsyujsj", "收到广播");
        f52289a0.startServices(context);
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/receiver/hhymfsyujsj.kt
package com.storm.safe.rock.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import androidx.work.ExistingWorkPolicy
import androidx.work.OneTimeWorkRequestBuilder
import androidx.work.WorkManager
import com.storm.safe.rock.keepalive.KeepAliveWorker

/**
 * JADX: hhymfsyujsj.java (93 lines) — HeartbeatAlarmReceiver
 * BroadcastReceiver that receives alarm/heartbeat broadcasts and starts services.
 *
 * Contains a companion object with scheduling helper methods:
 * - scheduleHeartbeat: schedule heartbeat alarm
 * - scheduleNextPing: schedule next ping (delegates to scheduleHeartbeat)
 * - scheduleRecovery: schedule recovery alarm (delegates to scheduleHeartbeat)
 * - scheduleDualRecovery: convenience for dual-mode recovery
 * - cancelPing / cancelRecovery: cancel scheduled alarms
 * - startServices: start keepalive service
 *
 * JADX references:
 * - al1.f43714a5 → keepalive service singleton
 * - t60 → logging utilities
 */
class hhymfsyujsj : BroadcastReceiver() {

    companion object {
        private const val TAG = "hhymfsyujsj"

        /**
         * Cancel pending ping alarm.
         * JADX: C0271a0.cancelPing
         */
        fun cancelPing(context: Context) {
            // JADX: body is just parameter null-check; actual cancel logic may have been stripped
        }

        /**
         * Cancel pending recovery alarm.
         * JADX: C0271a0.cancelRecovery
         */
        fun cancelRecovery(context: Context) {
            // JADX: body is just parameter null-check
        }

        /**
         * Schedule a heartbeat alarm.
         * JADX: C0271a0.scheduleHeartbeat
         *
         * @param context Application context
         * @param delayMs Delay in milliseconds (0 = immediate)
         */
        fun scheduleHeartbeat(context: Context, delayMs: Long = 0L) {
            // JADX: body is just parameter null-check — actual AlarmManager logic
            // may have been inlined or stripped by R8
            // vendor: scheduling body was empty in decompiled code
            // R8 likely inlined the AlarmManager logic or it was stripped
        }

        /**
         * Schedule next ping alarm. Delegates to scheduleHeartbeat.
         * JADX: C0271a0.scheduleNextPing
         */
        fun scheduleNextPing(context: Context, delayMs: Long = 0L) {
            scheduleHeartbeat(context, delayMs)
        }

        /**
         * Schedule recovery alarm. Delegates to scheduleHeartbeat.
         * JADX: C0271a0.scheduleRecovery
         *
         * @param context Application context
         * @param delayMs Delay in milliseconds
         * @param retryCount Current retry count (unused in decompiled body)
         */
        fun scheduleRecovery(context: Context, delayMs: Long, retryCount: Int = 0) {
            scheduleHeartbeat(context, delayMs)
        }

        /**
         * Schedule dual recovery (heartbeat + recovery).
         * JADX: C0271a0.scheduleDualRecovery
         */
        fun scheduleDualRecovery(context: Context) {
            scheduleHeartbeat(context, 0L)
        }

        /**
         * Start keepalive services.
         * JADX: C0271a0.startServices
         */
        fun startServices(context: Context) {
            try {
                // JADX: al1.f43714a5.getInstance(context).m209821a1()
                // Triggers keepalive service via WorkManager
                val request = OneTimeWorkRequestBuilder<KeepAliveWorker>().build()
                WorkManager.getInstance(context)
                    .enqueueUniqueWork("keepalive_trigger", ExistingWorkPolicy.KEEP, request)
                Log.d(TAG, "启动服务")
            } catch (e: Exception) {
                Log.e(TAG, "启动服务失败", e)
            }
            System.currentTimeMillis() // JADX: timestamp captured but unused
        }

        /**
         * Check if instance is the current registered companion.
         * Utility used in JADX for default parameter wrappers.
         */
        @JvmStatic
        fun isRunning(): Boolean = false
    }

    override fun onReceive(context: Context, intent: Intent) {
        Log.v(TAG, "收到广播")
        startServices(context)
    }
}
